CPP-Examples.cpp

```diff
@@ -11,16 +11,17 @@ int main()
 
    strcpy(dest, src); 
  
-   strncpy(dest,src, strlen(dest)); //DevSkim: ignore DS111237,DS154189 
+   strncpy(dest,src, strlen(dest)); 
    
-   strcat(dest, "this is bad");
+   strcat(dest, "this is bad");     //DevSkim: ignore all
 
    strncat(dest, "this is also bad", strlen(dest)); 
```

Comment:
[Banned C function detected (strncat)](https://github.com/Microsoft/DevSkim/blob/master/guidance/DS108330.md#ruleDS108330)
strncat adds the null terminator at character 'n + 1', rather than at the nth character. this frequently leads to the null terminator being added in the memory adjacent to the destination buffer, rather than in the destination buffer.

*Fix Guidance:* Use strcat_s or strlcat if possible. 


Suggestions:
[1] strcat_s(dest,  strlen(dest,  "this is also bad"));
[2] strlcat(dest,  "this is also bad",  strlen(dest));